lib/jxl/dec_render_pipeline.h

```diff
@@ -20,70 +20,187 @@ enum class RenderPipelineChannelMode {
   kIgnored = 0,
   kInPlace = 1,
   kInOut = 2,
+  kInput = 3,
```

Comment:
Would be nice to have some comments to document what these different ChannelModes mean (what assumptions or invariants they imply, etc).

Reply:
True! We can do that once we actually know what the assumptions and invariants are :P